### ISSUE TITLE
NoSuchMethodError

### ISSUE THREAD

**Issue by xiaotangfeidao** [deeplearning4j/deeplearning4j#3297]
#### Issue Description

I am getting the following error during program run
```
21:29:00.483 [main] DEBUG o.n.j.c.CudaAffinityManager - Manually mapping thread [21] to device [0], out of [1] devices...
21:29:00.483 [main] DEBUG o.n.j.c.CudaAffinityManager - Manually mapping thread [22] to device [0], out of [1] devices...
**Exception in thread "UniGC thread 0" java.lang.NoSuchMethodError: org.nd4j.linalg.memory.MemoryManager.isPeriodicGcActive()Z
	at org.nd4j.jita.allocator.impl.AtomicAllocator$UnifiedGarbageCollectorThread.run(AtomicAllocator.java:624)**

21:29:01.483 [main] DEBUG o.n.j.handler.impl.CudaZeroHandler - Creating bucketID: 5
21:29:01.499 [main] DEBUG o.n.j.handler.impl.CudaZeroHandler - Creating bucketID: 2
21:29:01.499 [main] DEBUG o.n.j.handler.impl.CudaZeroHandler - Creating bucketID: 1
21:29:01.499 [main] DEBUG o.n.j.handler.impl.CudaZeroHandler - Creating bucketID: 0
21:29:01.499 [main] DEBUG o.n.j.handler.impl.CudaZeroHandler - Creating bucketID: 4
**Exception in thread "main" java.lang.NoSuchMethodError: org.nd4j.linalg.jcublas.ops.executioner.CudaExecutioner.validateDataType**(Lorg/nd4j/linalg/api/buffer/DataBuffer$Type;Lorg/nd4j/linalg/api/ops/Op;)V
	at org.nd4j.linalg.jcublas.ops.executioner.CudaExecutioner.invoke(CudaExecutioner.java:1259)
	at org.nd4j.linalg.jcublas.ops.executioner.CudaGridExecutioner.pushToGrid(CudaGridExecutioner.java:215)
	at org.nd4j.linalg.jcublas.ops.executioner.CudaGridExecutioner.processAsGridOp(CudaGridExecutioner.java:297)
	at org.nd4j.linalg.jcublas.ops.executioner.CudaGridExecutioner.exec(CudaGridExecutioner.java:102)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.assign(BaseNDArray.java:1137)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.put(BaseNDArray.java:1889)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.putRow(BaseNDArray.java:3466)
	at org.nd4j.linalg.dataset.DataSet.merge2d(DataSet.java:260)
	at org.nd4j.linalg.dataset.DataSet.merge(DataSet.java:181)
	at org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator.next(RecordReaderDataSetIterator.java:186)
	at org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator.next(RecordReaderDataSetIterator.java:372)
	at org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator.next(RecordReaderDataSetIterator.java:52)
	at Yut_ML_Utilties.yut_DataSet.readCSVFile(yut_DataSet.java:104)
	at Neighborhood_Delineation.HKG_NB_Classification_v5.trainFirstStage(HKG_NB_Classification_v5.java:104)
	at Neighborhood_Delineation.HKG_NB_Classification_v5.main(HKG_NB_Classification_v5.java:200)
```
#### Version Information

* Deeplearning4j 0.8.0
* windows server 2012
* CUDA 8.0

Here is my pom file
```xml
    <dependency>
      <groupId>org.deeplearning4j</groupId>
      <artifactId>deeplearning4j-core</artifactId>
      <version>0.8.0</version>
    </dependency>


    <dependency>
      <groupId>org.nd4j</groupId>
      <artifactId>nd4j-cuda-8.0-platform</artifactId>
      <version>0.8.0</version>
    </dependency>

    <dependency>
      <groupId>org.deeplearning4j</groupId>
      <artifactId>deeplearning4j-cuda-8.0</artifactId>
      <version>0.8.0</version>
    </dependency>
```

**Comment by xiaotangfeidao** [comment]
Just to add, there is no issue with 0.7.2 builds

**Comment by saudet** [comment]
Looks like there is something importing old dependencies. Could we take a look at your `pom.xml` file and at the output of `mvn dependency:tree`? Thanks

**Comment by xiaotangfeidao** [comment]
Thanks saudet here is the complete pom.xml. 
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>fmac</groupId>
  <artifactId>yut</artifactId>
  <version>1.0-SNAPSHOT</version>
  <packaging>jar</packaging>

  <name>Neighborhood_Delineation</name>
  <url>http://maven.apache.org</url>

  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
  </properties>

  <dependencies>
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>3.8.1</version>
      <scope>test</scope>
    </dependency>


    <dependency>
      <groupId>org.deeplearning4j</groupId>
      <artifactId>deeplearning4j-core</artifactId>
      <version>0.8.0</version>
    </dependency>


    <dependency>
      <groupId>org.nd4j</groupId>
      <artifactId>nd4j-cuda-8.0-platform</artifactId>
      <version>0.8.0</version>
    </dependency>


    <dependency>
      <groupId>com.google.guava</groupId>
      <artifactId>guava</artifactId>
      <version>19.0</version>
    </dependency>

  </dependencies>
</project>

**Comment by xiaotangfeidao** [comment]
the dependency tree is pretty complicated

**Comment by xiaotangfeidao** [comment]
If I switch to 0.7.2 like below then everything is fine
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>fmac</groupId>
  <artifactId>yut</artifactId>
  <version>1.0-SNAPSHOT</version>
  <packaging>jar</packaging>

  <name>Neighborhood_Delineation</name>
  <url>http://maven.apache.org</url>

  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
  </properties>

  <dependencies>
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>3.8.1</version>
      <scope>test</scope>
    </dependency>


    <dependency>
      <groupId>org.deeplearning4j</groupId>
      <artifactId>deeplearning4j-core</artifactId>
      <version>0.7.2</version>
    </dependency>


    <dependency>
      <groupId>org.nd4j</groupId>
      <artifactId>nd4j-cuda-8.0-platform</artifactId>
      <version>0.7.2</version>
    </dependency>


    <dependency>
      <groupId>com.google.guava</groupId>
      <artifactId>guava</artifactId>
      <version>19.0</version>
    </dependency>

  </dependencies>
</project>

**Comment by saudet** [comment]
Thank you, it looks alright. It might be a corrupted local repository. Try to remove your `~/.m2/repository`. If that doesn't change anything, could you also provide a bit of sample code that isn't functioning?

**Comment by AlexDBlack** [comment]
Another possibility: this is being run through Eclipse perhaps? I've seen it cache old dependencies in the project, and not update them after changing versions in the pom.xml

**Comment by xiaotangfeidao** [comment]
I updated Maven to 3.5, was at 3.2 I think. Also there was a package that I manually included that had 0.7.2 dependencies which I removed.  I am not sure exactly what did it but the changes above fixed the issues. =)
